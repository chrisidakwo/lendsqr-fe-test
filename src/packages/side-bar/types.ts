import {FC} from "react";

import IconProps from "../../ui-kit/icon/IconProps";

export interface MenuItem {
    label: string;
    icon: FC<IconProps>;
    url?: string;
}
