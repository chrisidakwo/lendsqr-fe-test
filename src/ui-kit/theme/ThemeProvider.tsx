import React, {ReactNode} from 'react';
import {ThemeProvider as BaseThemeProvider} from 'styled-components';

import appTheme from './themes/appTheme';
import {defaultTheme} from "./defaultTheme";

export interface ThemeInterface {
  app: typeof appTheme;
}

export interface ThemeProps {
  theme: ThemeInterface;
}

export interface ThemeProviderProps {
  theme?: ThemeInterface;
  children?: ReactNode;
}

export const ThemeProvider = ({ theme = defaultTheme, children }: ThemeProviderProps): JSX.Element => {
  return <BaseThemeProvider theme={theme}>{children}</BaseThemeProvider>;
};
