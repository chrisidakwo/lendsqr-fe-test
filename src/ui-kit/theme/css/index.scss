@import "base/base";
@import "scaffolding";
@import "typography";
