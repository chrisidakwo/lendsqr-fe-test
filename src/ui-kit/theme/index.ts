import './css/index.scss';

export * from './defaultTheme';
export * from './definitions/breakpoints';
export * from './definitions/colors';
export { fontFamilies } from './definitions/typography';
export * from './ThemeProvider';
