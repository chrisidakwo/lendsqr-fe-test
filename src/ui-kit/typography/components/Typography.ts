import styled, {css} from 'styled-components';

import {getVariant} from './getVariant';
import {TypographyStylesProps, TypographyVariant} from '../Typography.types';
import {color, fontFamily, fontWeightBold, fontWeightMedium, fontWeightRegular, uppercase} from "./Typography.styles";

export interface TypographyProps {
    elementType: keyof JSX.IntrinsicElements;
    styleVariant?: string | TypographyVariant;
}

const commonStyles = css`
  font-family: ${({theme}): string => theme.app.typography.fontFamily};
  margin: 0;
  padding: 0;
`;

const Typography = styled.div.attrs(
    ({
         elementType,
     }: TypographyProps): {
        as?: keyof JSX.IntrinsicElements;
    } => ({as: elementType}),
)<TypographyProps & TypographyStylesProps>`
  ${commonStyles}

  ${({elementType, styleVariant}) => getVariant(elementType, styleVariant)}

  ${color}
  ${fontFamily}
  ${fontWeightRegular}
  ${fontWeightMedium}
  ${fontWeightBold}
  ${uppercase}
`;

export default Typography;
