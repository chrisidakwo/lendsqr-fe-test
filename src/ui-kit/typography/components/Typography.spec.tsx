import TestRenderer from "react-test-renderer";

import {ThemeProvider} from "../../theme";
import Typography from "./Typography";
import {TYPOGRAPHY_VARIANT_BODY_1, TYPOGRAPHY_VARIANT_BODY_2, TYPOGRAPHY_VARIANT_SUBSCRIPT} from "../Typography.types";

describe('UIKit/Typography', () => {
    describe('Typography', () => {
        it('correctly renders h1 text', () => {
            const renderer = TestRenderer.create(
                <ThemeProvider>
                    <Typography elementType={'h1'}>Lorem ipsum dolor</Typography>
                </ThemeProvider>,
            );
            const tree = renderer.toJSON();

            expect(renderer.root.findAllByType('h1')).toHaveLength(1);
            expect(tree).toHaveStyleRule('font-size', '2.25rem');
            expect(tree).toHaveStyleRule('font-weight', '600');
            expect(tree).toHaveStyleRule('line-height', '3rem');
        });

        it('renders typography as element with different variant styles', () => {
            const renderer = TestRenderer.create(
                <ThemeProvider>
                    <Typography elementType={'p'} styleVariant={'h1'} color={'hotpink'}>
                        Lorem ipsum dolor
                    </Typography>
                </ThemeProvider>,
            );
            const tree = renderer.toJSON();

            expect(renderer.root.findAllByType('h1')).toHaveLength(0);
            expect(renderer.root.findAllByType('p')).toHaveLength(1);
            expect(tree).toHaveStyleRule('font-size', '2.25rem');
            expect(tree).toHaveStyleRule('font-weight', '600');
            expect(tree).toHaveStyleRule('line-height', '3rem');
            expect(tree).toHaveStyleRule('color', 'hotpink');
        });

        it('renders typography with default styles', () => {
            const renderer = TestRenderer.create(
                <ThemeProvider>
                    <Typography elementType={'p'}>Lorem ipsum dolor</Typography>
                </ThemeProvider>,
            );
            const tree = renderer.toJSON();

            expect(tree).toHaveStyleRule('font-size', '1rem');
            expect(tree).toHaveStyleRule('line-height', '1.75rem');
        });
    });

    describe('variantCss', () => {
        it('renders variant styles', () => {
            const tree = TestRenderer.create(
                <ThemeProvider>
                    <div>
                        <Typography elementType={'div'} styleVariant={TYPOGRAPHY_VARIANT_BODY_1}>
                            Lorem ipsum dolor
                        </Typography>
                        <Typography elementType={'div'} styleVariant={TYPOGRAPHY_VARIANT_BODY_2}>
                            Lorem ipsum dolor
                        </Typography>
                        <Typography elementType={'p'} styleVariant={TYPOGRAPHY_VARIANT_SUBSCRIPT}>
                            Lorem ipsum dolor
                        </Typography>
                    </div>
                </ThemeProvider>,
            ).toJSON();

            expect(tree).toMatchSnapshot();
        });
    });

    describe('stylesCss', () => {
        it('renders color', () => {
            const tree = TestRenderer.create(
                <ThemeProvider>
                    <div>
                        <Typography
                            elementType={'div'}
                            styleVariant={TYPOGRAPHY_VARIANT_BODY_1}
                            color={'hotpink'}
                            fontWeightRegular
                        >
                            Lorem ipsum dolor
                        </Typography>
                        <Typography
                            elementType={'div'}
                            styleVariant={TYPOGRAPHY_VARIANT_BODY_1}
                            fontWeightMedium
                        >
                            Lorem ipsum dolor
                        </Typography>
                        <Typography
                            elementType={'div'}
                            styleVariant={TYPOGRAPHY_VARIANT_BODY_1}
                            fontWeightBold
                        >
                            Lorem ipsum dolor
                        </Typography>
                        <Typography elementType={'div'}>
                            Lorem ipsum dolor
                        </Typography>
                    </div>
                </ThemeProvider>,
            ).toJSON();

            expect(tree).toMatchSnapshot();
        });
    });
});
