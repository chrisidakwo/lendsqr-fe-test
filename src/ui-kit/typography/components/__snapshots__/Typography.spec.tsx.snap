// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIKit/Typography stylesCss renders color 1`] = `
.c0 {
  font-family: Work Sans,Segeo UI,Arial,Helvetica Neue,sans-serif;
  margin: 0;
  padding: 0;
  color: #213F7D;
  font-size: 1rem;
  line-height: 1.75rem;
  color: hotpink;
  font-weight: 400;
}

.c1 {
  font-family: Work Sans,Segeo UI,Arial,Helvetica Neue,sans-serif;
  margin: 0;
  padding: 0;
  color: #213F7D;
  font-size: 1rem;
  line-height: 1.75rem;
  font-weight: 500;
}

.c2 {
  font-family: Work Sans,Segeo UI,Arial,Helvetica Neue,sans-serif;
  margin: 0;
  padding: 0;
  color: #213F7D;
  font-size: 1rem;
  line-height: 1.75rem;
  font-weight: 700;
}

.c3 {
  font-family: Work Sans,Segeo UI,Arial,Helvetica Neue,sans-serif;
  margin: 0;
  padding: 0;
  color: #213F7D;
  font-size: 1rem;
  line-height: 1.75rem;
}

@media only screen and(max-width:767px) {
  .c0 {
    color: #545F7D;
    font-size: 0.85rem;
    line-height: 1.5rem;
  }
}

@media only screen and(max-width:767px) {
  .c1 {
    color: #545F7D;
    font-size: 0.85rem;
    line-height: 1.5rem;
  }
}

@media only screen and(max-width:767px) {
  .c2 {
    color: #545F7D;
    font-size: 0.85rem;
    line-height: 1.5rem;
  }
}

<div>
  <div
    className="c0"
    color="hotpink"
  >
    Lorem ipsum dolor
  </div>
  <div
    className="c1"
  >
    Lorem ipsum dolor
  </div>
  <div
    className="c2"
  >
    Lorem ipsum dolor
  </div>
  <div
    className="c3"
  >
    Lorem ipsum dolor
  </div>
</div>
`;

exports[`UIKit/Typography variantCss renders variant styles 1`] = `
.c0 {
  font-family: Work Sans,Segeo UI,Arial,Helvetica Neue,sans-serif;
  margin: 0;
  padding: 0;
  color: #213F7D;
  font-size: 1rem;
  line-height: 1.75rem;
}

.c1 {
  font-family: Work Sans,Segeo UI,Arial,Helvetica Neue,sans-serif;
  margin: 0;
  padding: 0;
  color: #545F7D;
  font-size: 0.875rem;
  line-height: 1.5rem;
}

.c2 {
  font-family: Work Sans,Segeo UI,Arial,Helvetica Neue,sans-serif;
  margin: 0;
  padding: 0;
  color: #545F7D;
  font-size: 0.85rem;
  line-height: 1.5rem;
}

@media only screen and(max-width:767px) {
  .c0 {
    color: #545F7D;
    font-size: 0.85rem;
    line-height: 1.5rem;
  }
}

<div>
  <div
    className="c0"
  >
    Lorem ipsum dolor
  </div>
  <div
    className="c1"
  >
    Lorem ipsum dolor
  </div>
  <p
    className="c2"
  >
    Lorem ipsum dolor
  </p>
</div>
`;
